### ISSUE TITLE
Startup script has to wait for service

### ISSUE THREAD

**Issue by potzkovge** [rancher/os#2031]
**RancherOS Version: (ros os version)**
v1.0.3
**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
baremetal, Dell PowerEdge r230

my cloud-config looks as follows:
[root@rancher ~]# cat /var/lib/rancher/conf/cloud-config.yml
rancher:
  services_include:
    kernel-headers-system-docker: true
    open-iscsi: true
  ssh: ...
write_files:
- content: |+
    #!/bin/bash
    echo "this was a triumph" > /tmp/test2.txt
    sleep 300
    iscsiadm ...
  owner: root
  path: /etc/rc.local
  permissions: "0755"

This works but only because i added a "sleep" command in my script which is a quite ugly solution. I want to use iscsiadm commands natively after the open-iscsi service is up and running. Is there any way for the startup scripts (rc.local or start.sh) to wait for the service? If not please implement a solution.

Regards,
Georgi

**Comment by SvenDowideit** [comment]
yes, `sleep` is the wrong way to do it :). The current "container" way, is to have a wait that polls to see if the other required containers are up and in a functioning state.

I would also not use a console based rc.local / start.sh script, I'd use another service that contains the script that then polls to see if open-iscsi is up.

In the future, I'm hoping we can add a "io.rancher.os.after_ready: open-iscsi" that would block the new service until the open-iscsi service reports its Healthcheck as ready.

At this point, I'm not sure what it will take to add this in to libcompose - but :+1: from me too.

**Comment by potzkovge** [comment]
@SvenDowideit Thank your for the fast response. Sounds good.
